earthfile2llb/converter.go

```diff
@@ -1645,16 +1647,21 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
 	runOpts = append(runOpts, llb.WithCustomNamef("%s%s", c.vertexPrefix(opts.Locally, isInteractive, false), commandStr))
 
 	var extraEnvVars []string
+
+	if c.ftrs.UseProjectSecrets && (c.varCollection.Org() == "" || c.varCollection.Project() == "") {
+		return pllb.State{}, errors.New("PROJECT statement is required with --use-project-secrets flag")
+	}
```

Comment:
I think this shouldn't be strictly required.

In the new implementation, it should be possible to use secrets only locally. When using secrets locally only, the user should not have to create an Earthly account and add the `PROJECT` declaration.

Reply:
I don't think we'll want to force the use of `PROJECT` once `--use-project-secrets` is no longer an optional feature flag. My thinking here was that we should probably enforce `PROJECT` if the feature flag is set for the time being. Once it's no longer optional, we can remove that check. I don't mind removing this check now if that's preferred.